src/mysky.ts

```diff
@@ -181,6 +187,43 @@ export class MySky {
     clearStoredSeed();
   }
 
+  /**
+   * signMessage will sign the given data using the MySky user's private key,
+   * this method can be used for MySky user verification as the signature may be
+   * verified against the user's public key, which is the MySky user id.
+   *
+   * NOTE: verifySignedMessage is the counter part of this method, and verifies
+   * the signed message and returns the message without signature
+   *
+   * NOTE: this function (internally) adds a salt to the given data array to
+   * ensure there's no potential overlap with anything else, like registry
+   * entries.
+   *
+   * @param message - message to sign
+   * @returns signed message
+   */
+  async signMessage(message: Uint8Array): Promise<Uint8Array> {
+    // fetch the user's seed
+    const seed = checkStoredSeed();
+    if (!seed) {
+      throw new Error("User seed not found");
+    }
+
+    // fetch the private key
+    const { privateKey, publicKey } = genKeyPairFromSeed(seed);
+    const privateKeyBytes = fromHexString(privateKey);
+    if (!privateKeyBytes) {
+      throw new Error("Corrupted key pair");
+    }
+
+    // prepend a salt to the message, essentially name spacing it so the
+    // signature is only useful for MySky ID verification
+    const hashed = new Uint8Array([...sha512(SALT_MYSKY_ID_VERIFICATION), ...message]);
```

Comment:
Maybe a question for David but why aren't we hashing the full message like we do elsewhere?

```suggestion
    const hashed = sha512(new Uint8Array([...sha512(SALT_MYSKY_ID_VERIFICATION), ...sha512(message)]));
```

We would need to pass the original message when verifying if we do this.

Reply:
Well, you hit the nail on the head here, I personally prefer to model it against the `tweetnacl` library interface. I don't like having to pass in the original message, you might not be able to predict it but be able to process it if it makes sense, right?